icicle-source/src/Icicle/Source/Parser.hs

```diff
@@ -37,6 +39,17 @@ parseQueryTop name inp
  = let toks = lexer "" inp
    in  runParser (consumeAll $ top name) () "" toks
 
+parseTop :: Text -> Either ParseError UnresolvedInputId
+parseTop inp
```

Comment:
can we call this "parseFactName" or "parseInputId" instead? the fact name is what makes it a top-level query, but the name isn't the top-level in itself

Reply:
`QueryTop` is probably a bad name too, maybe we should call it something else. shrug.